src/db.py

```diff
@@ -284,24 +286,38 @@ class TileData:
     source: str
     inactive_color: tuple[int, int]
     active_color: tuple[int, int]
-    tiling: int
+    tiling: TilingMode
     text_type: int
     text_direction: int | None
     tags: list[str]
+    extra_frames: list[int]
 
     @classmethod
     def from_row(cls, row: Row) -> TileData:
         """Create a tiledata object from a database row."""
+        tags: str | None = row["tags"]
+        if tags is not None:
+            tags = tags.strip()
+            if len(tags) == 0:
+                tags = None
+                
+        extra_frames: str | None = row["extra_frames"]
+        if extra_frames is not None:
+            extra_frames = extra_frames.strip()
+            if len(extra_frames) == 0:
+                extra_frames = None
+
         return TileData(
             row["name"],
             row["sprite"],
             row["source"],
             (row["inactive_color_x"], row["inactive_color_y"]),
             (row["active_color_x"], row["active_color_y"]),
-            row["tiling"],
+            TilingMode(row["tiling"]),
```

Comment:
An invalid tiling mode here seems like undefined behavior

Reply:
as long as what's in the database is fine then this is fine, but it probably would be better to add checking for it

Reply:
I strongly suggest adding checking here.